src/utils/strings.cpp

```diff
@@ -417,5 +417,14 @@ std::string string_md5(const char *buffer, unsigned length)
 
     return result;
 }
+
+std::string find_string_prefix(const std::string &input, char separator)
+{
+    std::size_t current = input.find(separator);
+    if (current == 0 || current == std::string::npos) {
+        return "";
```

Comment:
Initializing with c-style empty string literal may be less efficient for compilers with older version even if optimization is enabled. Use default constructor instead:
```suggestion
        return std::string();
```